File: 6/newton.py
import math

r = 2
a = 1
for i in range(5):
    z_x = a
    z_y = math.exp(a)

    s_a = z_y
    s_b = z_y - s_a*z_x

    a = (r-s_b) / s_a
    print(f'{i=}, {a=}')
print(math.log(2))

File: 6/newton.go
package main

import (
	"fmt"
	"math"
)

func main() {
	// y = log2の近似値を調べる(自然対数)
	r := 2.0
	a := 2.0

	for i := 0; i < 7; i++ {
		zahyoX := a
		zahyoY := math.Exp(a)

		sessenA := math.Exp(a)
		sessenB := zahyoY - sessenA * zahyoX

		nextA := (r - sessenB) / sessenA

		fmt.Printf("Step %d = %.15f -> %.15f\n", i+1, a, nextA)
		a = nextA
	}
	fmt.Println(1.0/3.0)
}
